### PR TITLE
net: openthread: allow to enable/disable TCP

### PR DESCRIPTION
**Pull request by edmont**
Allow to enable/disable the OpenThread TCP implementation. Disable it by default.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

### CODE REVIEW

File: subsys/net/lib/openthread/platform/openthread-core-zephyr-config.h [hunk 1]
```diff
@@ -401,4 +401,24 @@
...
 
+/**
+ * @def OPENTHREAD_CONFIG_TCP_ENABLE
+ *
+ * Enable TCP.
+ *
+ */
+#ifdef CONFIG_OPENTHREAD_TCP_ENABLE
+#define OPENTHREAD_CONFIG_TCP_ENABLE CONFIG_OPENTHREAD_TCP_ENABLE
+#endif /* CONFIG_OPENTHREAD_TCP_ENABLE */
```
Comment (pdgendt) [inline subsys/net/lib/openthread/platform/openthread-core-zephyr-config.h:412]: The default in `openthread` is to [enable it](https://github.com/zephyrproject-rtos/openthread/blob/zephyr/src/core/config/ip6.h#L186).

Maybe always define it?
```C
#define OPENTHREAD_CONFIG_TCP_ENABLE IS_ENABLED(CONFIG_OPENTHREAD_TCP_ENABLE)
```

File: subsys/net/l2/openthread/Kconfig.thread [hunk 2]
```diff
@@ -144,3 +144,10 @@ config OPENTHREAD_IP6_MAX_EXT_MCAST_ADDRS
 	int "The maximum number of supported IPv6 multicast addresses allows to be externally added"
 	range 0 32
 	default 2
+
+config OPENTHREAD_TCP_ENABLE
+	bool "Enable TCP"
+
+config OPENTHREAD_CLI_TCP_ENABLE
+	bool "Enable TCP in the CLI tool"
+	default y if OPENTHREAD_TCP_ENABLE
```
Comment (pdgendt) [inline subsys/net/l2/openthread/Kconfig.thread:153]:
```suggestion
config OPENTHREAD_CLI_TCP_ENABLE
	bool "Enable TCP in the CLI tool"
	default y if SHELL
	depends on OPENTHREAD_TCP_ENABLE
```

File: subsys/net/lib/openthread/platform/openthread-core-zephyr-config.h (hunk 1 above)
Comment (edmont) [reply subsys/net/lib/openthread/platform/openthread-core-zephyr-config.h]: Agree. Same goes for `OPENTHREAD_CONFIG_CLI_TCP_ENABLE`.

File: subsys/net/l2/openthread/Kconfig.thread (hunk 2 above)
Comment (edmont) [reply subsys/net/l2/openthread/Kconfig.thread]: Done.

**Review by pdgendt — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]

**Review by jfischer-no — APPROVED** [review]

**Pull request merged**